### DISCUSSION TITLE
Change Project.EntityFramework.Tests database provider

### DISCUSSION THREAD

**Discussion by LeanDualBoot** [abpframework/abp#20670] [Q&A] [answered]
Hi guys, I need to use HierarchyId for one of my projects and sice SQLite doesnt support it I need to change my database to be EntityFrameworkCore.InMemory.HierarchyId (this package containts the .UseInMemoryDatabase() used below)

```
public class TreeEntityFrameworkCoreTestModule : AbpModule
{
    public override void ConfigureServices(ServiceConfigurationContext context)
    {
        Configure<AbpDbContextOptions>(options => {
            options.Configure(context => {
                new InMdemoryTreeDbContext(new DbContextOptionsBuilder<InMdemoryTreeDbContext>().UseInMemoryDatabase("hierarchyId", x => x.UseHierarchyId()).Options);
            });
        });
    }
}
```

but when I run one of my tests this is the error I am getting:   System.InvalidOperationException : No database provider has been configured for this DbContext. A provider can be configured by overriding the 'DbContext.OnConfiguring' method or by using 'AddDbContext' on the application service provider. If 'AddDbContext' is used, then also ensure that your DbContext type accepts a DbContextOptions<TContext> object in its constructor and passes it to the base constructor for DbContext.

**Comment by maliming** [comment]
hi

- Your ABP Framework version.
- Your User Interface type (Angular/MVC/React... etc.) if the issue is related to a specific UI
- Exception message and stack trace if available (check the logs).
- Please share the steps and code to reproduce the problem.

**Comment by LeanDualBoot** [comment, marked as answer]
fixed it by removing the SQLIte dependency from EntityFrameworkCoreTestModule : AbpModule

```
[DependsOn(
    typeof(TreeApplicationTestModule),
    typeof(TreeEntityFrameworkCoreModule)
)]
public class TreeEntityFrameworkCoreTestModule : AbpModule
{
    public override void ConfigureServices(ServiceConfigurationContext context)
    {
        Configure<AbpDbContextOptions>(options => {
            options.Configure(context => {
                context.DbContextOptions.UseInMemoryDatabase("hierarchyId", x => x.UseHierarchyId());                
            });
        });
    }
}
```

dont forget to install EntityFrameworkCore.InMemory.HierarchyId
